Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2341,111 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-howto-row-factory:
+
+How to create and use row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represents each fetched row as a :class:`tuple`.
+If a :class:`!tuple` does not suit your needs,
+use the :class:`sqlite3.Row` class or a custom :attr:`~Connection.row_factory`.
+
+:class:`!Row` provides indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+In order to use :class:`!Row` as a row factory,
+simply assign it to the :attr:`Connection.row_factory` attribute:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+
+Query results are now returned as :class:`!Row` instances:
+
+.. doctest::
+
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0]  # Access by index.
+   'Earth'
+   >>> row["name"]    # Access by name.
+   'Earth'
+   >>> row["RADIUS"]  # Column names are case-insensitive.
+   6378
+
+It is also possible to assign row factories to cursors using
+:attr:`Cursor.row_factory`:
+
+.. doctest::
+
+   >>> cur = con.cursor()
+   >>> cur.row_factory == con.row_factory
+   True
+   >>> cur.row_factory = None  # Override cursor row factory.
+
+   # The cursor and the connection row factories now differ.
+   >>> cur.row_factory == con.row_factory
+   False
+   >>> cur.execute("SELECT 'Hello'").fetchone()
+   ('Hello',)
+
+   # The connection still uses sqlite3.Row, as set in the previous example.
+   >>> con.execute("SELECT 'Hello'").fetchone()
+   <sqlite3.Row object at ...>
+
+If more flexibility is needed, implement a custom row factory.
+Here's an example of one returning a :class:`dict`:
+
+.. doctest::
+
+   >>> def dict_factory(cursor, row):
+   ...     col_names = [col[0] for col in cursor.description]
+   ...     return {key: value for key, value in zip(col_names, row)}
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = dict_factory
+   >>> for row in con.execute("SELECT 1 AS a, 2 AS b"):
+   ...     print(row)
+   {'a': 1, 'b': 2}
+
+The following row factory returns a :class:`~collections.namedtuple`.
```

Comment:
Minot thing: in this instance, I would want to read `returns a named tuple`.

Contrary to one-word names like `defaultdict`, named tuple is a concept, or a type of class, not a specific class.
So I think ``` :class:`named tuple <collections.namedtuple>` ``` or ``` :term:`named tuple` ``` would read better here, and not be technically incorrect at all.

Reply:
Yup, great point. Following our previous discussion, this should instead point to the glossary entry you mentioned. Also, it should end with a colon rather than a period (full stop). Thus:

```suggestion
The following row factory returns a :term:`named tuple`:
```

Reply:
Not sure if this was missed, or was made locally but not pushed like the other changes?

Reply:
I was a little hesitant about "crying wolf" after the last two times, hehe